src/v/kafka/client/consumer.cc

```diff
@@ -310,10 +318,84 @@ consumer::offset_commit(std::vector<offset_commit_request_topic> topics) {
     return req_res(std::move(req_builder));
 }
 
-ss::future<shared_consumer_t>
-make_consumer(shared_broker_t coordinator, group_id group_id) {
+ss::future<fetch_response>
+consumer::fetch(std::chrono::milliseconds timeout, int32_t max_bytes) {
+    using broker_reqs_t = absl::node_hash_map<shared_broker_t, fetch_request>;
+
+    // Split requests by broker
+    broker_reqs_t broker_reqs;
+    for (auto const& [t, ps] : _assignment) {
+        for (const auto& p : ps) {
+            auto tp = model::topic_partition{t, p};
+            auto broker = co_await _brokers.find(tp);
+            auto& session = _fetch_sessions[broker];
+
+            auto& req = broker_reqs
+                          .try_emplace(
```

Comment:
shouldn't the try-emplace follow a vassert()

Reply:
The semantics are intentional. There is a hidden lazy_emplace which takes a functor that can defer construction, but it's awkward.

Reply:
what would be asserted?